bridges/snowbridge/pallets/inbound-queue-v2/src/api.rs

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+//! Implements the dry-run API.
+
+use crate::{Config, Error, Junction::AccountId32, Location};
+use snowbridge_core::inbound::Proof;
+use snowbridge_router_primitives::inbound::v2::{ConvertMessage, Message};
+use sp_core::H256;
+use xcm::latest::Xcm;
+
+pub fn dry_run<T>(message: Message, _proof: Proof) -> Result<Xcm<()>, Error<T>>
```

Comment:
Needs to return the fee too. Can use the weight for the `submit` extrinsic and use the `WeightToFee` converter to convert that into DOT.

Reply:
execution fee on BH as well as XCM prologue fee teleported to AH

Reply:
Addressed in https://github.com/claravanstaden/polkadot-sdk/pull/4/commits/42d688a28a31e76e58043955d3fbef2c5cb4a677.